{
    "bug_id": 48,
    "classification": {
        "singleLine": false
    },
    "commit": "f39bc60d",
    "failing_tests": [
        "org.apache.camel.processor.LogEipPropagateExceptionTest"
    ],
    "files": 1,
    "jira_id": "4388",
    "linesAdd": 7,
    "linesRem": 0,
    "nb_error": 0,
    "nb_failure": 1,
    "nb_skipped": 0,
    "nb_test": 3743,
    "patch": "diff --git a/camel-core/src/main/java/org/apache/camel/processor/LogProcessor.java b/camel-core/src/main/java/org/apache/camel/processor/LogProcessor.java\nindex 94a4246caf..ced8977ea8 100644\n--- a/camel-core/src/main/java/org/apache/camel/processor/LogProcessor.java\n+++ b/camel-core/src/main/java/org/apache/camel/processor/LogProcessor.java\n@@ -43,8 +43,15 @@ public void process(Exchange exchange) throws Exception {\n \n     @Override\n     public boolean process(Exchange exchange, AsyncCallback callback) {\n+        try {\n             String msg = expression.evaluate(exchange, String.class);\n             logger.log(msg);\n+        } catch (Exception e) {\n+            exchange.setException(e);\n+        } finally {\n+            // callback must be invoked\n+            callback.done(true);\n+        }\n         return true;\n     }\n \n",
    "project": "camel"
}